app/src/sharedTest/java/org/openobservatory/ooniprobe/InstantMessagingTestSuiteUtils.java

```diff
@@ -0,0 +1,65 @@
+package org.openobservatory.ooniprobe;
+
+import org.openobservatory.ooniprobe.test.test.AbstractTest;
+import org.openobservatory.ooniprobe.test.test.FacebookMessenger;
+import org.openobservatory.ooniprobe.test.test.Signal;
+import org.openobservatory.ooniprobe.test.test.Telegram;
+import org.openobservatory.ooniprobe.test.test.Whatsapp;
+
+import java.util.Arrays;
+import java.util.List;
+
+public class InstantMessagingTestSuiteUtils {
+
+    /**
+     * Saves a result in the DB and returns it with the given number of measurements, and
+     * all related model objects in the DB.
+     *
+     * @param accessibleMeasurements number of successful measurements
+     * @param accessibleTestTypes list with successful measurements
+     * @param blockedMeasurements  number of failed measurements
+     * @param blockedTestTypes list with failed measurements
+     * @throws IllegalArgumentException for excess number of measurements
+     */
+    public static void populateInstantMessagingMeasurements(
+            int accessibleMeasurements, List<AbstractTest> accessibleTestTypes,
+            int blockedMeasurements, List<AbstractTest> blockedTestTypes
```

Comment:
I do not like very much functions with mutable arguments, I believe they are a bit harder to understand and use. As an alternative, this function could return a struct with `accessibleTestTypes` and `blockedTestTypes` inside.